src/main/java/de/hysky/skyblocker/skyblock/waypoint/WaypointsListWidget.java

```diff
@@ -0,0 +1,291 @@
+                } else {
+                    entryIndex = WaypointsListWidget.this.children().indexOf(this) + 1;
+                    while (entryIndex < WaypointsListWidget.this.children().size() && !(WaypointsListWidget.this.children().get(entryIndex) instanceof WaypointCategoryEntry)) {
+                        entryIndex++;
+                    }
+                }
+                category.waypoints().add(waypointEntry.waypoint);
+                WaypointsListWidget.this.children().add(entryIndex, waypointEntry);
+            }).width(72).build();
+            buttonDelete = ButtonWidget.builder(Text.translatable("selectServer.deleteButton"), buttonDelete -> {
+                int entryIndex = WaypointsListWidget.this.children().indexOf(this) + 1;
+                while (entryIndex < WaypointsListWidget.this.children().size() && !(WaypointsListWidget.this.children().get(entryIndex) instanceof WaypointCategoryEntry)) {
+                    WaypointsListWidget.this.children().remove(entryIndex);
+                }
+                WaypointsListWidget.this.children().remove(this);
+                waypoints.remove(category);
+            }).width(38).build();
+            children = List.of(enabled, nameField, buttonNewWaypoint, buttonDelete);
+        }
+
+        @Override
+        public List<? extends Selectable> selectableChildren() {
+            return children;
+        }
+
+        @Override
+        public List<? extends Element> children() {
+            return children;
+        }
+
+        private void updateName(String name) {
+            int index = waypoints.indexOf(category);
+            category = category.withName(name);
+            if (index >= 0) {
+                waypoints.set(index, category);
+            }
+        }
+
+        @Override
+        public void render(DrawContext context, int index, int y, int x, int entryWidth, int entryHeight, int mouseX, int mouseY, boolean hovered, float tickDelta) {
+            enabled.setPosition(x, y + 1);
+            nameField.setPosition(x + 22, y);
+            buttonNewWaypoint.setPosition(x + entryWidth - 115, y);
+            buttonDelete.setPosition(x + entryWidth - 38, y);
+            for (ClickableWidget child : children) {
+                child.render(context, mouseX, mouseY, tickDelta);
+            }
+        }
+    }
+
+    protected class WaypointEntry extends AbstractWaypointEntry {
+        private final WaypointCategoryEntry category;
+        private NamedWaypoint waypoint;
+        private final List<ClickableWidget> children;
+        private final CheckboxWidget enabled;
+        private final TextFieldWidget nameField;
+        private final TextFieldWidget xField;
+        private final TextFieldWidget yField;
+        private final TextFieldWidget zField;
+        private final TextFieldWidget colorField;
+        private final ButtonWidget buttonDelete;
+
+        public WaypointEntry(WaypointCategoryEntry category) {
```

Comment:
Filling the coordinates with the player's coordinates could be also convenient

Reply:
Done in ee6664d.